apps/src/templates/instructions/codeReviewV2/CodeReviewTimelineElement.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {TextLink} from '@dsco_/link';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import color from '@cdo/apps/util/color';
+import javalabMsg from '@cdo/javalab/locale';
+
+export const codeReviewTimelineElementType = {
+  CREATED: 'created',
+  COMMIT: 'commit',
+  CODE_REVIEW: 'codeReview'
+};
+
+// This component represents elements on the code review and commit timeline. There are 3 types of elements:
+// 1. CREATED - this is the node that represents the first element on the timeline, it has a line extending below
+//    it unless it is also the only element in the timeline. Otherwise this node does not vary from project to project.
+// 2. COMMIT - this represents a commit to the project. It will display an eyeball on the left which links to the
+//    project version if the project version is present and not expired. It also handles a child which represents
+//    the specific content in the commit. If it is the last element in the timeline, it will not have a line extending
+//    down from it.
+// 3. CODE_REVIEW - this represnts a code review request and comments made. It will display an eyeball to the left
+//    which links to the project version if it is present and not expired. It takes a child which should be the actual
+//    code review component. If it is the last element in the timeline it will not have a line extending below it.
+const CodeReviewTimelineElement = ({
+  type,
+  isLast,
+  projectVersionId,
+  isProjectVersionExpired,
+  children
+}) => {
+  const displayVersion = !isProjectVersionExpired && !!projectVersionId;
+  const versionLink =
+    location.origin + location.pathname + '?version=' + projectVersionId;
+
+  if (type === codeReviewTimelineElementType.CREATED) {
+    return (
+      <div style={styles.element}>
+        <div style={styles.eyeColumn} />
+        <div style={styles.timeline}>
+          <TimelineDot color={color.purple} />
+          {!isLast && <TimelineLine height={40} marginRight={-1.5} />}
+        </div>
+        <div style={{...styles.child, ...styles.createdText}}>
+          {javalabMsg.created()}
+        </div>
+      </div>
+    );
+  }
+
+  if (type === codeReviewTimelineElementType.COMMIT) {
+    const borderLeft = isLast ? 'none' : lineStyle;
+
+    return (
+      <div style={styles.element}>
+        <div style={styles.eyeColumn}>
+          {displayVersion && <EyeballLink versionHref={versionLink} />}
+        </div>
+        <div style={styles.timeline}>
+          <TimelineDot color={color.dark_charcoal} hasCheck={true} />
```

Comment:
Why doesn't this type need a `TimelineLine` if it's not the last item?

Reply:
For the commit, the height of the timeline next to it is variable based on the height of the commit, so instead of having a line coming down from it, I have a border-left on the commit so that it will grow as the commit grows. Maybe I should add a comment about this because it was also confusing to James

https://user-images.githubusercontent.com/24235215/165986499-711d3f44-44c4-4364-a677-11be38a083b6.mov

Reply:
If we wanted to use a TimelineLine here instead of a border left, is the problem that we don't know what to set the height of the TimelineLine to?  Would it be possible to use something like "100%" to have the height match the height of the commit (so that we use TimelineLine to draw all of the lines)?

In general, I think the confusion here is that the line is drawn two different ways and the relationship between those two is not super clear.  I'm ok with it as-is (but would appreciate a comment), just throwing out some suggestions for what might make this clearer.

Reply:
Good point, I'll see if I can make the TimelineLine work in this scenario as well

Reply:
Ok, I've updated it using the TimelineLine instead of the border left which makes way more sense and is consistent with the other elements, thanks for pushing on this!

Reply:
Looks great, thanks for taking another look!